go/pkg/log/repository/log.go

```diff
@@ -88,6 +88,21 @@ func (r *LogRepository) PullRecords(ctx context.Context, collectionId string, of
 		Limit:        int32(batchSize),
 		Timestamp:    timestamp,
 	})
+	// Relies on the fact that the records are ordered by offset.
+	if len(records) > 0 && records[0].Offset != offset {
+		trace_log.Error("Error in pulling records from record_log table. Some entries have been purged.", zap.String("collectionId", collectionId), zap.Int("requestedOffset", int(offset)), zap.Int("actualOffset", int(records[0].Offset)))
+		records, err = nil, errors.New("some entries have been purged")
+	}
+	// This means that the log is empty i.e. compaction_offset = enumeration_offset
+	// AND also all records have been purged. In this case, if the requested offset
+	// is less than the compacted offset (or enumeration offset), we should return an error.
+	if len(records) == 0 {
+		var compacted_offset, offset_err = r.GetLastCompactedOffsetForCollection(ctx, collectionId)
```

Comment:
what happens if offset_err is not nil - I think this err is silently ignored then?